dataframely/failure.py

```diff
@@ -102,15 +98,25 @@ def scan_parquet(cls, source: str | Path | IO[bytes]) -> Self:
             The failure info object.
         """
         lf = pl.scan_parquet(source)
-        # NOTE: In `write_parquet`, the rule columns are added as the name of the last
-        #  column.
-        last_column = lf.collect_schema().names()[-1]
-        metadata = json.loads(last_column)
-        rule_columns = metadata["rule_columns"]
-        *schema_module_parts, schema_name = metadata["schema"].split(".")
+
+        # We can read the rule columns either from the metadata of the Parquet file
+        # or, to remain backwards-compatible, from the last column of the lazy frame if
+        # the parquet file is missing metadata.
+        if (meta := pl.read_parquet_metadata(source).get("dataframely")) is not None:
+            metadata = json.loads(meta)
+            rule_columns: list[str] = metadata["rule_columns"]
+            schema_name: str = metadata["schema"]
+        else:  # pragma: no cover
+            last_column = lf.collect_schema().names()[-1]
+            metadata = json.loads(last_column)
+            rule_columns = metadata["rule_columns"]
+            schema = metadata["schema"]
+            lf = lf.drop(last_column)
+
+        *schema_module_parts, schema_name = schema_name.split(".")
```

Comment:
```suggestion
            schema_name = metadata["schema"]
            lf = lf.drop(last_column)

        *schema_module_parts, schema_name = schema_name.split(".")
```

Shouldn't it be like this? Have you tested if the compatibility code works?